server/graphql/types/src/types/purchase_order_line.rs

```diff
@@ -0,0 +1,109 @@
+use async_graphql::{dataloader::DataLoader, *};
+use chrono::NaiveDate;
+use graphql_core::{loader::ItemLoader, standard_graphql_error::StandardGraphqlError, ContextExt};
+use repository::{ItemRow, PurchaseOrderLine, PurchaseOrderLineRow};
+use service::{usize_to_u32, ListResult};
+
+use crate::types::ItemNode;
+
+#[derive(PartialEq, Debug)]
+pub struct PurchaseOrderLineNode {
+    pub purchase_order_line: PurchaseOrderLineRow,
+    pub item: ItemRow,
+}
+
+#[derive(SimpleObject)]
+pub struct PurchaseOrderLineConnector {
+    pub total_count: u32,
+    pub nodes: Vec<PurchaseOrderLineNode>,
+}
+
+#[Object]
+impl PurchaseOrderLineNode {
+    pub async fn id(&self) -> &str {
+        &self.row().id
+    }
+    pub async fn purchase_order_id(&self) -> &str {
+        &self.row().purchase_order_id
+    }
+    pub async fn line_number(&self) -> &i32 {
+        &self.row().line_number
+    }
+    pub async fn item(&self, ctx: &Context<'_>) -> Result<ItemNode> {
+        let loader = ctx.get_loader::<DataLoader<ItemLoader>>();
+
+        let result = loader.load_one(self.item.id.to_string()).await?;
+
+        result.map(ItemNode::from_domain).ok_or(
+            StandardGraphqlError::InternalError(format!(
+                "Cannot find item ({}) linked to purchase_order_line ({})",
+                &self.item.id,
+                &self.row().id
+            ))
+            .extend(),
+        )
+    }
```

Comment:
I wasn't quite sure why - but existing pattern is to use loader here for querying lines rather than mapping directly from service layer even if the service layer contains the item line.

You can see this for example in stock_line (also invoice_line and requisition_line) where the struct contains an itemRow, but we still use a loader in graphql to take the id from that struct, and then load through a loader.

Is there some optimisation this allows that I am not aware of?

I wasn't sure - but copied this pattern in anycase.